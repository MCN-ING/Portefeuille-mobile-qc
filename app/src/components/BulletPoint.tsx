import { useTheme } from '@hyperledger/aries-bifold-core'
import { StyleSheet, Text, View, ViewStyle } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

type Props = {
  text: string
  style?: ViewStyle
}

const BulletPoint = ({ text, style }: Props) => {
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'flex-start',
    },
    iconContainer: {
      marginRight: 10,
      marginVertical: 10,
    },
    bulletText: {
      ...TextTheme.normal,
      textAlign: 'left',
    },
  })
  return (
    <View style={[styles.container, style]}>
      <View style={styles.iconContainer}>
        <Icon name={'circle'} size={6} color={ColorPallet.brand.modalIcon} />
      </View>
      <Text style={styles.bulletText}>{text}</Text>
    </View>
  )
}

export default BulletPoint
