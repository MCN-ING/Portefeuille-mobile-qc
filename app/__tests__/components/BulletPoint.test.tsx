import { render } from '@testing-library/react-native'
import React from 'react'
import BulletPoint from '../../src/components/BulletPoint'

// Mock the RNPermissionsModule
jest.mock('react-native-permissions', () => ({
  check: jest.fn(),
  request: jest.fn(),
}))

describe('BulletPoint Component', () => {
  test('Renders correctly', async () => {
    const tree = render(<BulletPoint text={'Any Text'} />)

    expect(tree).toMatchSnapshot()
  })
})
