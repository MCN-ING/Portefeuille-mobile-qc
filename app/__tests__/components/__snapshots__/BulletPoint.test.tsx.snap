// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPoint Component Renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "marginRight": 10,
        "marginVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 6,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î½Š
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 18,
        "fontWeight": "normal",
        "textAlign": "left",
      }
    }
  >
    Any Text
  </Text>
</View>
`;
